src/back/backend_handler.rs

```diff
@@ -222,15 +223,29 @@ impl BackEndHandlerBuilder {
         self
     }
 
-    pub fn build(self) -> BackEndHandler {
-        BackEndHandler {
-            provider: self.provider.expect("Provider missing"),
-            converter: self.converter.expect("Converter missing"),
-            provider_id: self.provider_id.expect("Provider ID missing"),
-            content_type: self.content_type.expect("Content type missing"),
-            accept_type: self.accept_type.expect("Accept type missing"),
-            version_min: self.version_min.expect("Version min missing"),
-            version_maj: self.version_maj.expect("Version maj missing"),
-        }
+    pub fn build(self) -> std::io::Result<BackEndHandler> {
+        Ok(BackEndHandler {
+            provider: self
+                .provider
+                .ok_or_else(|| Error::new(ErrorKind::InvalidData, "provider is missing"))?,
```

Comment:
If you don't also log something in the lambda, `ok_or(Error::new(ErrorKind::InvalidData, "provider is missing"))?` is simpler

Reply:
I tried that but clippy complains:
```rust
error: use of `ok_or` followed by a function call
   --> src/front/domain_socket.rs:148:48
    |
148 |           DomainSocketListener::new(self.timeout.ok_or(
    |  ________________________________________________^
149 | |             Error::new(
150 | |                 ErrorKind::InvalidInput,
151 | |                 "listener timeout missing",
152 | |                 )
153 | |         )?)
    | |_________^
    |
    = note: `-D clippy::or-fun-call` implied by `-D clippy::all`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#or_fun_call
help: try this
    |
148 |         DomainSocketListener::new(self.timeout.ok_or_else(|| Error::new(
149 |                 ErrorKind::InvalidInput,
150 |                 "listener timeout missing",
151 |                 ))?)
    |
```
It also makes it easier to maintain and add a log in teh future :)

Reply:
Fair enough, this makes sense! Thanks for the explanation!